tabular/src/autogluon/tabular/models/tabpfnmix/tabpfnmix_model.py

```diff
@@ -0,0 +1,221 @@
+from __future__ import annotations
+
+import os
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+
+from autogluon.common.utils.pandas_utils import get_approximate_df_mem_usage
+from autogluon.common.utils.resource_utils import ResourceManager
+from autogluon.common.utils.try_import import try_import_torch
+from autogluon.core.constants import BINARY, MULTICLASS, REGRESSION, QUANTILE
+from autogluon.core.models import AbstractModel
+from autogluon.features.generators import LabelEncoderFeatureGenerator
+
+
+# TODO: Add huggingface weights download support
+class TabPFNMixModel(AbstractModel):
+    """
+    [Experimental model] Can be changed/removed without warning in future releases.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._feature_generator = None
+
+    def _get_model_type(self):
+        from ._internal.tabpfnmix_classifier import TabPFNMixClassifier
+        if self.problem_type in ['binary', 'multiclass']:
+            model_cls = TabPFNMixClassifier
+        else:
+            # FIXME: Add regression support
+            raise AssertionError(f"TabPFN does not support problem_type='{self.problem_type}'")
+        return model_cls
+
+    def _set_default_params(self):
+        """Specifies hyperparameter values to use by default"""
+        default_params = {
+            "n_ensembles": 1,
+            "max_epochs": 0,
+            "split_val": False,
+            "use_best_epoch": True,
+        }
+        for param, val in default_params.items():
+            self._set_default_param_value(param, val)
+
+    # FIXME: Make X_val, y_val = None work well, currently it uses X and y as validation, should instead skip validation entirely
+    # FIXME: Use `params_trained` for refit optimal epochs
+    # FIXME: Handle model weights download
+    # FIXME: GPU support?
+    # FIXME: Save model weights to file instead of pickling?
+    def _fit(self, X: pd.DataFrame, y: pd.Series, X_val: pd.DataFrame = None, y_val: pd.Series = None, num_cpus: int = 1, num_gpus: float = 0, **kwargs):
+        try_import_torch()
+        import torch
+        from ._internal.config.config_run import ConfigRun
+
+        ag_params = self._get_ag_params()
+        max_classes = ag_params.get("max_classes")
+        if max_classes is not None and self.num_classes > max_classes:
+            # TODO: Move to earlier stage when problem_type is checked
+            raise AssertionError(f"Max allowed classes for the model is {max_classes}, " f"but found {self.num_classes} classes.")
+
+        params = self._get_model_params()
+
+        # FIXME: Don't require loading from file, allow user to specify everything?
+        # TODO: Not a big fan of the path config logic due to possible portability issues
+        if "path_config" in params:
+            path_config = Path(params["path_config"])
```

Comment:
Do we actually need this branch? I couldn’t find `path_config` referenced anywhere else in the code. If it’s not in use, I think it’s safe to remove it to avoid redundancy

Reply:
I ended up completely removing this logic.